src/transport/raw/TCP.cpp

```diff
@@ -626,7 +626,15 @@ CHIP_ERROR TCPBase::DoHandleIncomingConnection(const Inet::TCPEndPointHandle & l
                                                uint16_t peerPort)
 {
     PeerAddress addr;
-    ReturnErrorOnFailure(GetPeerAddress(*endPoint, addr));
+    CHIP_ERROR getPeerError = GetPeerAddress(*endPoint, addr);
+    // See https://github.com/project-chip/connectedhomeip/issues/41746
+    // Failures here must be handled carefully so that broken connections
+    // continue to propagate to failure callbacks to avoid flaky tests
+    if (getPeerError != CHIP_NO_ERROR)
+    {
+        SuccessOrLog(getPeerError, "Failure getting peer info, using fallback");
```

Comment:
we know it is a failure, how about direct ChipLogError and not using SuccessOrLog?

Reply:
Mostly because `SuccessOrLog` automatically takes care of suffixing with `% CHIP_ERROR_FORMAT` & makes this a bit nicer to read 😅🤣

I could add yet another macro... but not sure that's worthwhile (& `getPeerError` being a local var, the compiler should be able to optimize away the extra conditional)

Reply:
My readability complaint is that this adds an extra layer of indirection for readability:
  - successorlog is another layer in front of "log"
  - needs me to reason about "this is clearly an error, so this will log unconditionally"